File: mo_lucro_app/lib/utils/formatters.dart
import 'package:intl/intl.dart';

/// Centralized formatters for the app.
class AppFormatters {
  AppFormatters._();

  static final _brl = NumberFormat.currency(
    locale: 'pt_BR',
    symbol: 'R\$',
    decimalDigits: 2,
  );

  static final _percent = NumberFormat('+##0.00;-##0.00');
  static final _dateShort = DateFormat('dd/MM/yy', 'pt_BR');
  static final _dateFull = DateFormat('dd MMM yyyy', 'pt_BR');
  static final _month = DateFormat('MMMM yyyy', 'pt_BR');

  static String currency(double value) => _brl.format(value);

  static String currencyCompact(double value) {
    if (value.abs() >= 1000000) {
      return 'R\$ ${(value / 1000000).toStringAsFixed(1)}M';
    }
    if (value.abs() >= 1000) {
      return 'R\$ ${(value / 1000).toStringAsFixed(1)}K';
    }
    return _brl.format(value);
  }

  static String percent(double value) => '${_percent.format(value)}%';

  static String percentSimple(double value) =>
      '${value.toStringAsFixed(2)}%';

  static String dateShort(DateTime date) => _dateShort.format(date);
  static String dateFull(DateTime date) => _dateFull.format(date);
  static String month(DateTime date) => _month.format(date);
}

File: mobile/src/utils/formatters.ts
import { format } from 'date-fns';
import { ptBR } from 'date-fns/locale';

/**
 * Porte de lib/utils/formatters.dart.
 *
 * Observação sobre separadores: no app Flutter só as datas usavam locale pt_BR
 * (initializeDateFormatting). Os NumberFormat de percentual não recebiam locale,
 * então caíam no padrão en_US e imprimiam ponto decimal. O comportamento foi
 * mantido igual aqui para a migração não mudar o que a tela mostra hoje.
 */

const brl = new Intl.NumberFormat('pt-BR', {
  style: 'currency',
  currency: 'BRL',
  minimumFractionDigits: 2,
  maximumFractionDigits: 2,
});

export function currency(value: number): string {
  return brl.format(value);
}

export function currencyCompact(value: number): string {
  const abs = Math.abs(value);
  if (abs >= 1_000_000) return `R$ ${(value / 1_000_000).toFixed(1)}M`;
  if (abs >= 1_000) return `R$ ${(value / 1_000).toFixed(1)}K`;
  return brl.format(value);
}

/** Sempre com sinal explícito — equivale ao padrão '+##0.00;-##0.00'. */
export function percent(value: number): string {
  const sign = value < 0 ? '-' : '+';
  return `${sign}${Math.abs(value).toFixed(2)}%`;
}

export function percentSimple(value: number): string {
  return `${value.toFixed(2)}%`;
}

/** dd/MM/yy */
export function dateShort(date: Date): string {
  return format(date, 'dd/MM/yy', { locale: ptBR });
}

/** dd MMM yyyy */
export function dateFull(date: Date): string {
  return format(date, 'dd MMM yyyy', { locale: ptBR });
}

/** MMMM yyyy */
export function month(date: Date): string {
  return format(date, 'MMMM yyyy', { locale: ptBR });
}
